src/node.js

```diff
@@ -173,6 +175,21 @@
     }
   }
 
+  startup.setupProcessObject = function() {
+    process._setupProcessObject(setPropByIndex);
+
+    function setPropByIndex(val0, val1, val2, val3, val4, val5, val6, val7) {
+      this.push(val0);
+      if (val1) this.push(val1);
+      if (val2) this.push(val2);
+      if (val3) this.push(val3);
+      if (val4) this.push(val4);
+      if (val5) this.push(val5);
+      if (val6) this.push(val6);
+      if (val7) this.push(val7);
```

Comment:
+1 on refactoring this to use arguments.length... this `val0, ... val7` stuff gives me the willies.